src/StreamingDataFrames/streamingdataframes/models/serializers/quix.py

```diff
@@ -132,91 +213,96 @@ def __call__(self, value: bytes, ctx: SerializationContext) -> Iterable[dict]:
-            "TagValues": {"tag1": ["tag"]}
-        }
+        yield from self._deserializers[model_key](value=deserialized)
 
-        Output:
-        {
-            "a": "str",
-            "b": 1,
-            "c": 1.1,
-            "d": b"bytes",
-            "Tags": {"tag1": "tag"}},
-            "__Q_Timestamp": 123123123
+    def _is_legacy_format(self, value: bytes):
+        try:
+            if value.startswith(b"<"):
+                raise SerializationError(
+                    "Message splitting was detected in a legacy-formatted message; "
+                    "it is not supported in the new client."
+                )
+            return value.startswith(b'{"C":') and b'"K":' in value
+        except AttributeError:
+            return False
+
+    def _parse_legacy_format(
+        self, value: bytes
+    ) -> Tuple[Union[Mapping, List[Mapping]], Mapping]:
+        value = self._loads(value, **self._loads_kwargs)
+        headers = {
+            QCodecId.HEADER_NAME: value.pop(LegacyHeaders.Q_CODEC_ID),
+            QModelKey.HEADER_NAME: value.pop(LegacyHeaders.Q_MODEL_KEY),
         }
-    """
+        message = value.pop("V")
+        return headers, message
 
-    def deserialize(
-        self, model_key: str, value: Union[List[Mapping], Mapping]
-    ) -> Iterable[Mapping]:
-        # Fail if model key is unsupported
-        if model_key not in (QModelKey.TIMESERIESDATA, QModelKey.PARAMETERDATA):
-            raise SerializationError(
-                f'"{QModelKey.HEADER_NAME}" must be "{QModelKey.TIMESERIESDATA}" '
-                f'or {QModelKey.PARAMETERDATA}, got"{model_key}"'
-            )
-        if not isinstance(value, Mapping):
-            raise SerializationError(f'Expected mapping, got type "{type(value)}"')
 
-        timestamps = value["Timestamps"]
-        # Make a list of parameters and iterators with values to get them one by one
-        # and decode values from base64 if they're binary
-        all_params = [
-            (
-                param_name,
-                iter(param_values),
-                True if param_type == "BinaryValues" else False,
-            )
-            for param_type in ("NumericValues", "StringValues", "BinaryValues")
-            for param_name, param_values in value.get(param_type, {}).items()
-        ]
-        # Do the same with TagValues
-        tags = [
-            (tag, iter(values)) for tag, values in value.get("TagValues", {}).items()
-        ]
+class QuixLegacySerializer(JSONSerializer):
+    extra_headers = {}
+    _legacy = {}
 
-        # Iterate over timestamps and get a corresponding parameter value
-        # for each timestamp
-        for timestamp_ns in timestamps:
-            row_value = {
-                param: _b64_decode_or_none(next(values)) if is_binary else next(values)
-                for param, values, is_binary in all_params
+    def __init__(
+        self,
+        as_legacy: Optional[bool] = None,
```

Comment:
Why to use `Optional[bool]` here when it can be just `True` or `False`?

Reply:
Yeah true, I can def do that cleaner.